src/mobster/cmd/generate/oci_image/__init__.py

```diff
@@ -0,0 +1,113 @@
+"""A module for generating SBOM documents for OCI images."""
+
+__all__ = ["GenerateOciImageCommand"]
+
+import json
+import logging
+from pathlib import Path
+from typing import Any
+
+from spdx_tools.spdx.jsonschema.document_converter import DocumentConverter
+from spdx_tools.spdx.model.document import Document
+from spdx_tools.spdx.parser.jsonlikedict.json_like_dict_parser import JsonLikeDictParser
+from spdx_tools.spdx.writer.write_utils import convert
+
+from mobster.cmd.generate.base import GenerateCommandWithOutputTypeSelector
+from mobster.cmd.generate.oci_image.add_image import extend_sbom_with_image_reference
+from mobster.cmd.generate.oci_image.base_images_dockerfile import (
+    extend_sbom_with_base_images_from_dockerfile,
+)
+from mobster.cmd.generate.oci_image.cyclonedx_wrapper import CycloneDX1BomWrapper
+from mobster.cmd.generate.oci_image.spdx_utils import normalize_sbom
+from mobster.image import Image
+from mobster.sbom.merge import merge_sboms
+
+LOGGER = logging.getLogger(__name__)
+
+
+class GenerateOciImageCommand(GenerateCommandWithOutputTypeSelector):
+    """
+    Command to generate an SBOM document for an OCI image.
+    """
+
+    @staticmethod
+    async def _dump_sbom_to_dict(
+        sbom: Document | CycloneDX1BomWrapper,
+    ) -> dict[str, Any]:
+        if isinstance(sbom, Document):
+            return convert(sbom, DocumentConverter())  # type: ignore[no-untyped-call]
+        return sbom.to_dict()
+
+    # pylint: disable=too-many-locals
+    async def execute(self) -> Any:
+        """
+        Generate an SBOM document for OCI image.
+        """
+        LOGGER.debug("Generating SBOM document for OCI image")
+        # Argument parsing
+        syft_boms: list[Path] = self.cli_args.from_syft or []
+        hermeto_bom: Path = self.cli_args.from_hermeto
+        image_pullspec: str = self.cli_args.image_pullspec
+        image_digest: str = self.cli_args.image_digest
+        parsed_dockerfile_path: Path = self.cli_args.parsed_dockerfile_path
+        dockerfile_target_stage: str = self.cli_args.dockerfile_target
+        additional_base_images: list[str] = self.cli_args.additional_base_image or []
+        # contextualize: bool = self.cli_args.contextualize
+        # TODO add contextual SBOM utilities    # pylint: disable=fixme
+        # pylint: enable=fixme
+
+        with open(parsed_dockerfile_path, encoding="utf-8") as parsed_dockerfile_io:
```

Comment:
The dockerfile argument is not required in the CLI interface and this line fails. We should process it only when the option is provided and based on that we should augment the sbom. The same should be applied to other optional arguments if there are any.